tsl/src/nodes/vector_agg/exec.c

```diff
@@ -0,0 +1,204 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+
+#include <postgres.h>
+
+#include <commands/explain.h>
+#include <executor/executor.h>
+#include <nodes/extensible.h>
+#include <nodes/makefuncs.h>
+#include <nodes/nodeFuncs.h>
+
+#include "exec.h"
+
+#include "compression/arrow_c_data_interface.h"
+#include "nodes/decompress_chunk/compressed_batch.h"
+#include "nodes/decompress_chunk/exec.h"
+#include "guc.h"
+#include "functions.h"
+
+static void
+vector_agg_begin(CustomScanState *node, EState *estate, int eflags)
+{
+	CustomScan *cscan = castNode(CustomScan, node->ss.ps.plan);
+	node->custom_ps =
+		lappend(node->custom_ps, ExecInitNode(linitial(cscan->custom_plans), estate, eflags));
+}
+
+static void
+vector_agg_end(CustomScanState *node)
+{
+	ExecEndNode(linitial(node->custom_ps));
+}
+
+static void
+vector_agg_rescan(CustomScanState *node)
+{
+	if (node->ss.ps.chgParam != NULL)
```

Comment:
Could we have a test case that performs a rescan? I added this [test code](https://github.com/timescale/timescaledb/blob/52094a3103a19d0ccb77fa88eeb4867df40a2634/tsl/test/sql/compression.sql#L903) some time ago to force a rescan of a DeconpressChunk node. Maybe you can adapt it for an aggregation query.

Reply:
I have a test for this in `vector_agg_param`, will check why it doesn't have the coverage.

Reply:
I made some changes to improve coverage. There are still some uncovered lines left, I'd keep them to be on the safe side, because here we are working with any possible Postgres plan, e.g. from other extensions, so it's better to be accepting.